File: functions_array/SingleNumber.java
class SingleNumber {
    public static void main(String args[]) {
        int arr[]={1,2,1,2,4};
        int ans = 0;
        int  n=arr.length;
        for(int i =0;i<n;i++){
            ans = ans ^ arr[i];
        }
        System.out.println("The number repeating only once is: "+ans);
    }
}

File: functions_array/Single_Number.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums){
        int ans=0,n=nums.size();
        for(int i=0;i<n;i++){
            ans=ans^nums[i];
        }
        return ans;
    }
};

int main(){
    vector<int> nums={4,1,2,1,2};

    Solution s;
    cout<<s.singleNumber(nums);
    return 0;
}